packages/assets-controllers/src/TokenListController.ts

```diff
@@ -286,80 +291,67 @@ export class TokenListController extends StaticIntervalPollingController<TokenLi
   async fetchTokenList(chainId: Hex): Promise<void> {
     const releaseLock = await this.mutex.acquire();
     try {
-      const { tokensChainsCache } = this.state;
-      let tokenList: TokenListMap = {};
-      // Attempt to fetch cached tokens
-      const cachedTokens = await safelyExecute(() =>
-        this.#fetchFromCache(chainId),
+      if (this.isCacheValid(chainId)) {
+        return;
+      }
+
+      // Fetch fresh token list from the API
+      const tokensFromAPI = await safelyExecute(
+        () =>
+          fetchTokenListByChainId(
+            chainId,
+            this.abortController.signal,
+          ) as Promise<TokenListToken[]>,
       );
-      if (cachedTokens) {
-        // Use non-expired cached tokens
-        tokenList = { ...cachedTokens };
-      } else {
-        // Fetch fresh token list from the API
-        const tokensFromAPI = await safelyExecute(
-          () =>
-            fetchTokenListByChainId(
+
+      // Have response - process and update list
+      if (tokensFromAPI) {
+        // Format tokens from API (HTTP) and update tokenList
+        const tokenList: TokenListMap = {};
+        for (const token of tokensFromAPI) {
+          tokenList[token.address] = {
+            ...token,
+            aggregators: formatAggregatorNames(token.aggregators),
+            iconUrl: formatIconUrlWithProxy({
               chainId,
-              this.abortController.signal,
-            ) as Promise<TokenListToken[]>,
-        );
-
-        if (tokensFromAPI) {
-          // Format tokens from API (HTTP) and update tokenList
-          tokenList = {};
-          for (const token of tokensFromAPI) {
-            tokenList[token.address] = {
-              ...token,
-              aggregators: formatAggregatorNames(token.aggregators),
-              iconUrl: formatIconUrlWithProxy({
-                chainId,
-                tokenAddress: token.address,
-              }),
-            };
-          }
-        } else {
-          // Fallback to expired cached tokens
-          tokenList = { ...(tokensChainsCache[chainId]?.data || {}) };
+              tokenAddress: token.address,
+            }),
+          };
         }
+
+        this.update((state) => {
+          state.tokensChainsCache[chainId] ??= {
+            data: {},
+            timestamp: Date.now(),
+          };
```

Comment:
I didn't like all the spreads we were doing, so wanted to just focus on using immer.
However this also looks a little confusing...

I'll separate this out into a more readable object `defaultTokensCache`.